docker/service/src/main/java/uk/nhs/adaptors/scr/config/ExceptionHandlerFilter.java

```diff
@@ -33,22 +34,34 @@
     @Override
     public void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain chain)
         throws ServletException, IOException {
-        var token = request.getHeader(CORRELATION_ID_HEADER);
-        if (StringUtils.isNotEmpty(token) && !token.matches(UUID_REGEX)) {
+        var correlationIdToken = request.getHeader(CORRELATION_ID_HEADER);
+        var loggingIdToken = request.getHeader(REQUEST_ID_LOGGER);
 
-            response.setStatus(HttpStatus.BAD_REQUEST.value());
-            response.setContentType(APPLICATION_FHIR_JSON_VALUE);
-
-            var operationOutcome = new OperationOutcome();
-            operationOutcome.addIssue()
-                .setCode(OperationOutcome.IssueType.EXCEPTION)
-                .setSeverity(OperationOutcome.IssueSeverity.ERROR)
-                .setDiagnostics("Invalid " + CORRELATION_ID_HEADER + ". Should be a UUIDv4 matching \"" + UUID_REGEX + "\"")
-                .setDetails(new CodeableConcept().addCoding(NHSCodings.BAD_REQUEST.asCoding()));
-
-            response.getWriter().write(fhirParser.encodeResource(APPLICATION_FHIR_JSON, operationOutcome));
+        if (checkValidUUID(correlationIdToken)) {
+            throwInvalidUUIDResponse(response, CORRELATION_ID_HEADER);
+        } else if (checkValidUUID(loggingIdToken)) {
+            throwInvalidUUIDResponse(response, REQUEST_ID_LOGGER);
         } else {
             chain.doFilter(request, response);
         }
     }
+
+    public boolean checkValidUUID(String header) {
```

Comment:
If you check something, you would expect that it would return `False` if data is not correct. Here you return `True`